devel/nodejs14/Portfile

```diff
@@ -82,35 +82,19 @@ post-patch {
 }
 
 # use the system libuv instead of the bundled version, as it is fixed for older systems
-if { ${os.platform} eq "darwin" && ${os.major} < 11 } {
-    # on macOS nodejs only builds against libc++
-    # this force is OK as node does not link against any other c++ libs
-    # per @kencu, libcxx is only for 10.5 and 10.6 as a prerequisite for an even-newer libcxx (macports-libcxx)
-    depends_lib-append      port:libcxx
+if { ${os.platform} eq "darwin" && ${os.major} < 15 } {
     depends_lib-append    path:lib/libuv.dylib:libuv
     configure.args-append --shared-libuv
-}
-
-#apparently even macos14 needs the flags that i originally included for older macs.
-#so i am going to make the 'ignore warnings as errors' applicable for all targets
-configure.cppflags-append   -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
-configure.cflags-append     -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
-configure.cxxflags-append   -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
-
-if { ${os.platform} eq "darwin" && ${os.major} < 15 } {
-    # macports-libcxx is required for all platforms below 10.13, i think.
-    # set it to brosemite and lower (14). increase to 18 if it indeed is up to high sierra.
-    depends_lib-append          port:macports-libcxx
-    depends_build-append        port:macports-libcxx
-    configure.ldflags-append    -L/opt/local/lib/libcxx
-    # as @kencu has stated on the mailing-list, it is preferred to simply add the line 
-    # 
-    # legacysupport.use_mp_libcxx   yes 
-    #
-    # however the stock compiler for 10.7 etc is too old
-    # unless @kencu wants to fix the clash between macports-libcxx headers
-    # and each mp-clang, this will be the solution. i don't see it as problematic.
 
+    # suppress a warning-as-error that shows up with newer clang compilers
+    if {[string match *clang* ${configure.compiler}]} {
+        configure.cxx-append      -Wno-error=enum-constexpr-conversion
+    }
+    
+    if {${configure.cxx_stdlib} eq "libc++"} {
+        depends_lib-append        port:macports-libcxx
+        configure.cxx-append      -Wl,-L${prefix}/lib/libcxx
```

Comment:
You probably meant `configure.cxxflags-append`? This is likely `ldflag` rather than `cxxflag`, though I can imagine that a build system may not recognize ldflags correctly.

Reply:
> You probably meant `configure.cxxflags-append`? This is likely `ldflag` rather than `cxxflag`, though I can imagine that a build system may not recognize ldflags correctly.

Yes, this looks incorrect as-is.

Reply:
cxxflags does not work, see here: https://github.com/macports/macports-ports/pull/26031#issuecomment-2502191543